src/apps/Altinn.AccessManagement/src/Altinn.AccessManagement/HostedServices/Services/AllAltinnRoleSyncService.cs

```diff
@@ -0,0 +1,227 @@
+                                batchData.Add(assignment.Asignment);
+                            }                            
+                        }
+                        else
+                        {
+                            var filter = _assignmentRepository.CreateFilterBuilder();
+                            filter.Equal(t => t.FromId, assignment.Asignment.FromId);
+                            filter.Equal(t => t.ToId, assignment.Asignment.ToId);
+                            filter.Equal(t => t.RoleId, assignment.Asignment.RoleId);
+                            await _assignmentRepository.Delete(filter, curretOptions);
+                        }
+
+                        previousOptions = curretOptions;                        
+                    }
+                }
+
+                if (batchData.Count > 0)
+                {
+                    await Flush(batchId);
+                }
+
+                if (string.IsNullOrEmpty(page?.Content?.Links?.Next))
+                {
+                    return;
+                }
+
+                await UpdateLease(ls, data => data.AllAltinnRoleStreamNextPageLink = page.Content.Links.Next, cancellationToken);
+                
+                async Task Flush(Guid batchId)
+                {
+                    try
+                    {
+                        _logger.LogInformation("Ingest and Merge Assignment batch '{0}' to db", batchName);
+                        var ingested = await _ingestService.IngestTempData<Assignment>(batchData, batchId, previousOptions, cancellationToken);
+
+                        if (ingested != batchData.Count)
+                        {
+                            _logger.LogWarning("Ingest partial complete: Assignment ({0}/{1})", ingested, batchData.Count);
+                        }
+
+                        var merged = await _ingestService.MergeTempData<Assignment>(batchId, previousOptions, GetAssignmentMergeMatchFilter, cancellationToken: cancellationToken);
+
+                        _logger.LogInformation("Merge complete: Assignment ({0}/{1})", merged, ingested);
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogError(ex, "Failed to ingest and/or merge Assignment and EntityLookup batch {0} to db", batchName);
+                        throw new Exception(string.Format("Failed to ingest and/or merge Assignment and EntityLookup batch {0} to db", batchName), ex);
+                    }
+                    finally
+                    {
+                        batchId = Guid.CreateVersion7();
+                        batchData.Clear();
+                    }
+                }
+            }
+        }
+
+        private async Task<(Assignment Asignment, ChangeRequestOptions Options)> ConvertRoleDelegationModelToAssignment(RoleDelegationModel model)
+        {
+            try
+            {
+                var role = await GetOrCreateRole(model.RoleTypeCode);
```

Comment:
Unsure of we should use GetOrCreateRole here or rely on that the role must have been created through our migration script..